### ISSUE TITLE
Containers spinned up from recent BC images do not load all symbols

### ISSUE THREAD

**Issue by rvanbekkum** [microsoft/navcontainerhelper#550]
### Summary
For one of our projects we run automatic builds against a Docker container. Recently we pulled the latest Docker image for BC 14.3 (because we use some of the newly introduced events), but ever since our build script uses this new image the compilation step _infrequently_ fails due to it missing symbols for some of the objects from the base application.
Sometimes the build is succesful, but most of the time it fails since we pulled the new image.
The objects it reports as missing also slightly differ with each build.
The build script has not given us any issues with symbol loading up until we pulled the new image to compile our app against.

![screenshot_failedCompile](https://user-images.githubusercontent.com/7383241/62561209-f22df900-b87e-11e9-8730-8fa3d73387e3.png)
![screenshot_failedCompile2](https://user-images.githubusercontent.com/7383241/62562706-ed1e7900-b881-11e9-90ac-1d33e54782b3.png)

### Details
We use the following for our build:
* Build script found at: https://dev.azure.com/businesscentralapps/HelloWorld/_git/HelloWorld
* Image (for at least all of the following the problem occurs):
  * mcr.microsoft.com/businesscentral/onprem:w1-ltsc2016 (N.B., 14.3, i.e., the current latest)
  * mcr.microsoft.com/businesscentral/onprem:nl-ltsc2016
  * mcr.microsoft.com/businesscentral/sandbox:w1-ltsc2016
* NavContainerHelper 0.6.2.91 (automatically installed by build script)
* The container is created as specified in the `Create-Container.ps1` script, i.e.:
```powershell
New-NavContainer @parameters `
                 -containerName $containerName `
                 -imageName $imageversion.containerImage `
                 -auth NAVUserPassword `
                 -Credential $credential `
                 -alwaysPull:$imageversion.alwaysPull `
                 -updateHosts `
                 -includeCSide `
                 -doNotExportObjectsToText `
                 -enableSymbolLoading `
                 -useBestContainerOS `
                 -shortcuts $shortcuts `
                 -licenseFile "$licenseFile" `
                 -additionalParameters $additionalParameters `
                 -myScripts $myscripts
```

The output from the `Create Build Container` step for one of the builds is as follows:
```
##[section]Starting: Create Build Container
==============================================================================
Task         : PowerShell
Description  : Run a PowerShell script on Linux, macOS, or Windows
Version      : 2.151.2
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/powershell
==============================================================================
Generating script.
Formatted command: . 'D:\agent\BA01\_work\3\s\scripts\Create-Container.ps1' -version "current" -credential ([PSCredential]::new("navadmin", (ConvertTo-SecureString -String "***" -AsPlainText -Force))) -licenseFile "C:\temp\license.flf"
========================== Starting Command Output ===========================
##[command]"C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe" -NoLogo -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command ". 'D:\agent\BA01\_work\_temp\48b1a398-8248-4038-8478-63569ed99a6b.ps1'"
Create BC-588-bld from mcr.microsoft.com/businesscentral/onprem:w1-ltsc2016
NavContainerHelper is version 0.6.2.91
Host is Microsoft Windows Server 2016 Standard - ltsc2016
Docker Client Version is 18.09.6
Docker Server Version is 18.09.6
Using image mcr.microsoft.com/businesscentral/onprem:w1-ltsc2016
Creating Container BC-588-bld
Using license file C:\temp\license.flf
Version: 14.3.34444.0-W1
Platform: 14.0.34251.0
Generic Tag: 0.0.9.9
Container OS Version: 10.0.14393.3025 (ltsc2016)
Host OS Version: 10.0.14393.2972 (ltsc2016)
WARNING: The container operating system matches the host operating system, but the revision is different.
If you encounter issues, you might want to specify -isolation hyperv
Using locale en-US
Using process isolation
Disabling the standard eventlog dump to container log every 2 seconds (use -dumpEventLog to enable)
Files in C:\ProgramData\NavContainerHelper\Extensions\BC-588-bld\my:
- AdditionalOutput.ps1
- AdditionalSetup.ps1
- license.flf
- MainLoop.ps1
- updatehosts.ps1
Creating container BC-588-bld from image mcr.microsoft.com/businesscentral/onprem:w1-ltsc2016
3838212d7bd9c4d766e90ea89857800d514ec70253e9e278b8439a795d2165ec
Waiting for container BC-588-bld to be ready
Initializing...
Starting Container
Hostname is BC-588-bld
PublicDnsName is BC-588-bld
Using NavUserPassword Authentication
Starting Local SQL Server
Starting Internet Information Server
Creating Self Signed Certificate
Self Signed Certificate Thumbprint 78E492F70672B12348DEB671317B78FA03CFFD29
Modifying Service Tier Config File with Instance Specific Settings
Starting Service Tier
Registering event sources
Creating DotNetCore Web Server Instance
Using license file 'c:\run\my\license.flf'
Import License
Creating http download site
Setting SA Password and enabling SA
Creating navadmin as SQL User and add to sysadmin
Creating SUPER user
Container IP Address: 172.20.164.1
Container Hostname  : BC-588-bld
Container Dns Name  : BC-588-bld
Web Client          : http://BC-588-bld/NAV/
Dev. Server         : http://BC-588-bld
Dev. ServerInstance : NAV

Files:
http://BC-588-bld:8080/al-3.0.150570.vsix

Initialization took 99 seconds
Ready for connections!
Reading CustomSettings.config from BC-588-bld
Container BC-588-bld successfully created
##[section]Finishing: Create Build Container
```

### Additional Information
I originally posted a message on the dynamicsnavdev Yammer on this issue, where the following question was also asked:

> Are you creating your container with enablesymbolloading?

_Yes_, we create the container with the `-enableSymbolLoading` flag. That is how the `New-NavContainer` cmdlet is called in the `Create-Container.ps1` script.
**Notice:** When we remove `-enableSymbolLoading` then the compile step runs succesfully _consistently_ and the error of incomplete symbols never occurs.
The `-enableSymbolLoading` is however needed to load the Test Toolkit symbols in the `Import TestToolkit` step (which again is needed to compile the test app).

**Comment by freddydk** [comment]
Enablesymbolloading is not needed for testtoolkit.
https://freddysblog.com/2019/03/16/symbols-demystified/
remove enablesymbolloading and create a dependency to Test in your app.json.

You state that it infrequently fails the build - not consistently - this makes me think that it isn't a problem building the images, but rather a timing issue during build - will investigate to see if I can repro the problem

**Comment by rvanbekkum** [comment]
Ah, yes of course, if you remove -enableSymbolLoading then you still get the symbols if you add the "test" property in your test app's `app.json` file.
We removed the "test" property due to the symbols being loaded on import of the test toolkit again.
Thank you for that suggestion, it is a much better solution anyway since we only use the standard application, system and test symbols to compile our project.

It is still strange that not all standard symbols are loaded when the `-enableSymbolLoading` flag is added though.

**Comment by freddydk** [comment]
I think it is a timing issue.

If I wait for 10 minutes after creating the container before I build the app - then the symbols seems to be fine. If I do it immediately after they are (sometimes) cribbled.

Maybe something like the service tier is working on symbol generation and will abort this when you download them.

**Comment by freddydk** [comment]
All tests i have done shows that the way import-testtoolkit uses generatesymbols doesn't work in 14.x - not sure the reason for this, but I will change the way to import the objects and then call generate-symbols afterwards. This will take more time, but work.

**Comment by freddydk** [comment]
Fixed in 0.6.2.92
